frontend/packages/console-shared/src/components/actions/ActionsHookResolver.tsx

```diff
@@ -0,0 +1,32 @@
+import * as React from 'react';
+import { ExtensionHook, Action } from '@console/dynamic-plugin-sdk';
+
+type ActionsHookResolverProps = {
+  id: string;
+  scope: any;
+  useValue: ExtensionHook<Action[]>;
+  onValueResolved: (value: Action[], id: string) => void;
+  onValueError: (error: any) => void;
+};
+
+const ActionsHookResolver: React.FC<ActionsHookResolverProps> = ({
+  id,
+  useValue,
+  scope,
+  onValueResolved,
+  onValueError,
+}) => {
+  const [value, loaded, loadError] = useValue(scope);
+
+  React.useEffect(() => {
+    if (loaded) onValueResolved(value, id);
+  }, [id, loaded, onValueResolved, value]);
```

Comment:
IMO it makes sense to only execute this callback if loaded and the value changes.
Not sure you need the ID. Seems like that could be something handled by the parent? But I suppose you can keep it if needed.
```suggestion
  React.useEffect(() => {
    if (loaded) onValueResolved(value);
  }, [loaded, value]);
```

Reply:
Yeah, the ID is needed to create a map of extension ids to the actions returned by the provider. Also, do we really need to remove `onValueResolved` from the dependencies? It's a callback from the parent which doesn't change after the first render anyway.

Reply:
If the parent changes `onValueResolved` you don't actually want to re-run the effect. You only care to notify when `value` changes.

The reason I said ignore the `id` prop is because the parent can do this and keep the `id` in the closure.
`onValueResolved={(value) => onValueResolved(value, id)}`

Reply:
Updated. Seems like it will have the same effect though, earlier `onValueResolved` never changed after first render but now it does and we removed it from the dependencies array explicitly.

Reply:
Yes the outcome is the same :)
But this is a pattern that should be applied in other components in future. You shouldn't rely on a stable reference of the callback to ensure we don't over-notify. When used in an effect in this manner, only the change in the trigger value should result in a notification and not a change in the callback itself.